pydicom/dataset.py

```diff
@@ -2419,8 +2421,9 @@ def from_json(
         -------
         Dataset
         """
-        if not isinstance(json_dataset, dict):
-            json_dataset = json.loads(json_dataset)
+        if isinstance(json_dataset, str):
```

Comment:
Was there a reason this was `not isinstance(..., dict)` rather than `isinstance(..., str)`?